toolchain/sem_ir/function.h

```diff
@@ -52,14 +76,30 @@ struct Function : public Printable<Function> {
   // The return type. This will be invalid if the return type wasn't specified.
   TypeId return_type_id;
   // The storage for the return value, which is a reference expression whose
-  // type is the return type of the function. Will be invalid if the function
-  // doesn't have a return slot. If this is valid, a call to the function is
-  // expected to have an additional final argument corresponding to the return
-  // slot.
-  InstId return_slot_id;
+  // type is the return type of the function. This may or may not be used by the
+  // function, depending on whether the return type needs a return slot.
+  InstId return_storage_id;
   // Whether the declaration is extern.
   bool is_extern;
 
+  // The following member is set on the first call to the function, or at the
+  // point where the function is defined.
+
+  // Whether the function uses a return slot.
+  enum class ReturnSlot : int8_t {
```

Comment:
Declaration order. I know mixed feelings on this, but I think we should be following the style guide in spots like this.

https://google.github.io/styleguide/cppguide.html#Declaration_Order